src/usvfs_dll/sharedparameters.cpp

```diff
@@ -198,6 +200,46 @@ bool SharedParameters::executableBlacklisted(
   return false;
 }
 
+void SharedParameters::addSkipFileSuffix(const std::string& fileSuffix)
+{
+  bi::scoped_lock lock(m_mutex);
+
+  m_fileSuffixSkipList.insert(shared::StringT(fileSuffix.begin(), fileSuffix.end(),
+                                              m_fileSuffixSkipList.get_allocator()));
+}
+
+void SharedParameters::clearSkipFileSuffixes()
+{
+  bi::scoped_lock lock(m_mutex);
+  m_fileSuffixSkipList.clear();
+}
+
+std::vector<std::string> SharedParameters::skipFileSuffixes() const
```

Comment:
I don't really understand why this returns a new vector - This seems pretty costly as this needs to create a vector (although of small size) for every linked file/directory.

Do you assume the underlying list could be modified while inside a call to VirtualLinkXXX?

Reply:
It would create a new vector only once in `VirtualLinkDirectoryStatic` which from my understanding is intended to be used when you're dealing with one gigantic directory (like a Game)

`VirtualLinkFile` would create a new vector every time a **specific** file is linked, from my understanding `VirtualLinkFile` would be called if you only want to link ONE file externally. `VirtualLinkFile` isn't used in `VirtualLinkDirectoryStatic`

Reply:
Also, yes I opted to copy the internal list into a new vector since this class could be accessed and used by simultaneous threads.